docs/source/hyperparameters.rst

```diff
@@ -105,21 +105,18 @@ modify the network and read those values in the LightningModule
 
     class LitMNIST(LightningModule):
 
-        def __init__(self, hparams):
+        def __init__(self, layer_1_dim):
```

Comment:
this can be dangerous is any argument matches class method/property

Reply:
I guess this will always be a problem with this approach. The user will have to learn the LightningModule and not name the hparams after attributes. 

I think we should throw an error in this case. Otherwise the users code will not work anyway and they will have to find the bug.

Reply:
yeah, @awaelchli good point. Let's throw an error if it's a defined (ie: protected) name in the lightningModule

Reply:
I have added log warning, or you prefer another?